src/Kunstmaan/MediaBundle/Helper/File/FileHandler.php

```diff
@@ -61,13 +60,18 @@ class FileHandler extends AbstractMediaHandler
      *
      * @param int                              $priority
      * @param MimeTypeGuesserFactoryInterface  $mimeTypeGuesserFactory
-     * @param ExtensionGuesserFactoryInterface $extensionGuesserFactoryInterface
+     * @param ExtensionGuesserFactoryInterface|null $extensionGuesserFactoryInterface
      */
-    public function __construct($priority, MimeTypeGuesserFactoryInterface $mimeTypeGuesserFactory, ExtensionGuesserFactoryInterface $extensionGuesserFactoryInterface)
+    public function __construct($priority, MimeTypeGuesserFactoryInterface $mimeTypeGuesserFactory, $extensionGuesserFactoryInterface = null)
     {
         parent::__construct($priority);
         $this->mimeTypeGuesser = $mimeTypeGuesserFactory->get();
-        $this->extensionGuesser = $extensionGuesserFactoryInterface->get();
+
+        $this->extensionGuesser = new MimeTypes();
+        if ($extensionGuesserFactoryInterface instanceof ExtensionGuesserFactoryInterface) {
+            $this->extensionGuesser = $extensionGuesserFactoryInterface->get();
+            @trigger_error('Passing the "@kunstmaan_media.extension_guesser.factory" service as third argument is deprecated since KunstmaanMediaBundle 5.6 and will be replaced by MimeTypes in KunstmaanMediaBundle 6.0. Do not inject anything instead.', E_USER_DEPRECATED);
```

Comment:
```suggestion
            @trigger_error('Passing the "@kunstmaan_media.extension_guesser.factory" service as third argument is deprecated since KunstmaanMediaBundle 5.6 and will be replaced by the "@mime_types" service in KunstmaanMediaBundle 6.0. Inject the correct service instead.', E_USER_DEPRECATED);
```